matrix_common/versionstring.py

```diff
@@ -29,6 +29,7 @@
 version_cache: Dict[str, str] = {}
 
 
+@functools.lru_cache
```

Comment:
I have a sneaky suspicion that this form is only valid on Python > 3.8

See: [`lru_cache`](https://docs.python.org/dev/library/functools.html#functools.lru_cache):

> Changed in version 3.8: Added the *user_function* option.

Reply:
Oh I see---did you always have to provide a maxsize beforehand?

Reply:
I think you can just put parens on the end, `lru_cache()` isn't the same as `lru_cache` (one of the really confusing bits of decorators).

Reply:
I guess we don't have tests for this unfortunately. 😢

Reply:
We'll have one soon!